flake.nix

```diff
@@ -3,16 +3,26 @@
 
   inputs.flake-utils.url = "github:numtide/flake-utils";
 
-  outputs = { self, nixpkgs, flake-utils }: (flake-utils.lib.eachDefaultSystem (system: {
-    packages =  let
-      pkgs = import nixpkgs { inherit system; overlays = [ self.overlay ]; };
-    in {
+  outputs = { self, nixpkgs, flake-utils }: (flake-utils.lib.eachDefaultSystem (system: let
+    pkgs = import nixpkgs { inherit system; overlays = [ self.overlay ]; };
+  in {
+    packages = {
       inherit (pkgs.haskellPackages) Agda;
 
       # TODO agda2-mode
     };
 
     defaultPackage = self.packages.${system}.Agda;
+
+    devShell = pkgs.mkShell {
+      inputsFrom = [ self.defaultPackage.${system} ];
+      packages = with pkgs; [
```

Comment:
Unfortunately, I cannot judge the contribution to the nix flake.
Maybe it would be better to keep concerns separated and make a separate PR for the nix stuff.
Or find someone that approves these changes to the nix flake.

Reply:
Moved to https://github.com/agda/agda/pull/5870